actioncable/lib/action_cable/channel/test_case.rb

```diff
@@ -62,6 +62,25 @@ def initialize(identifiers = {})
       def transmit(cable_message)
         transmissions << cable_message.with_indifferent_access
       end
+
+      def connection_identifier
+        unless defined? @connection_identifier
+          @connection_identifier = connection_gid identifiers.filter_map { |id| instance_variable_get("@#{id}") }
```

Comment:
Which instance variable is this reading? I don't see any instance variable for the identifier being set in this object. Should not it be calling the method that was defined?

Reply:
This is copied verbatim from `ActionCable::Connection::Identification`.

I don't have any particular attachment to this implementation. Indeed, when I adapted it for our library, the end result was this being added to our `test_helper.rb`:
```rb
module ActionCable
  module Channel
    class ConnectionStub
      def connection_identifier
        connection_gid identifiers.map { |id| instance_variable_get("@#{id}") }.compact
      end

      def connection_gid(ids)
        ids.map { |o| o.respond_to?(:to_gid_param) ? o.to_gid_param : o.to_s }.sort.join(":")
      end
    end
  end
end
```
I used the code from `ActionCable::Connection::Identification` because I figured that it would have the lowest friction.

Reply:
In the identifications module, they are defined in https://github.com/rails/rails/blob/a07d0680787ced3c04b362fa7a238c918211ac70/actioncable/lib/action_cable/connection/identification.rb#L21.

I don't see that call in the ConnectionStub class.

Reply:
Thanks Rafael - you are right, of course. I needed to come back to this with a clear head and compare what's in `ActionCable::Connection::Identification`, what's in this PR and what's in our library.

I pushed a commit that swaps out the incorrect `instance_variable_get("@#{id}")` and instead it now `send`s the identifiers so that `connection_gid` has objects to operate on.

Let me know if you have any suggestions that might improve upon this approach. In the meantime, I believe that it is now behaving as expected.

Reply:
There is a ConnectionStub test.
https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/actioncable/test/channel/test_case_test.rb#L55-L66
A test should be added for this as well I think.

Reply:
Let me know if https://github.com/rails/rails/pull/42574/commits/b18e6ba6bd1596cb3ffbae05ad3c4680284baa13 is sufficient.